src/gmail.js

```diff
@@ -961,6 +961,26 @@ var Gmail = function(localJQuery) {
             if (pathname && (pathname.endsWith("/i/s") || pathname.endsWith("/i/fd"))) {
                 api.tools.parse_request_payload(params, triggered);
             }
+            if (pathname && (pathname.endsWith("/i/s"))) {
+                if (params.body_params && params.body_params[2] && params.body_params[2][1] && params.body_params[2][1][0][2][2][7]) {
+                    // check params length  and get thread ids
+                    // console.log('params.body_params: ', params.body_params);
+                    if (params.body_params[2][1][0][2][2][7][2].length > 3) {
+                        const newThreadIds = params.body_params[2][1].map(obj => obj[2][1]);
+                        const newMessageIds = params.body_params[2][1].map(obj => obj[2][2][7][3]);
+                        // console.log(newMessageIds);
+                        // console.log('newThreadIds: ', newThreadIds);
```

Comment:
These commented out logging statements should probably be removed once we have code in a finalized form which we see work well.